src/avro/src/schema.rs

```diff
@@ -2839,13 +2876,8 @@ mod tests {
 
         let schema = Schema::from_str(raw_schema).unwrap();
         assert_eq!(
-            "5ecb2d1f0eaa647d409e6adbd5d70cd274d85802aa9167f5fe3b73ba70b32c76",
+            "c4d97949770866dec733ae7afa3046757e901d0cfea32eb92a8faeadcc4de153",
```

Comment:
Is this not expected to be stable across different Avro libraries? If yes, I guess the old code was just computing the wrong fingerprint?

Reply:
It was computing something which... I interpret to be correct, but the Python avro library's canonical_form doesn't match. Now they do match, which is probably better.

It's still busted, though, on more complicated schemas, (it doesn't actually fill in the fullnames for things that didn't have a `namespace` entry but was getting them from the enclosing schema), but fixing that can be a follow-up.

Reply:
(It's no _worse_ than it was before, which is why I want to fix it in a separate PR)

Reply:
Yeah, moving in the right direction sounds great!